credential/base/src/main/java/org/wildfly/security/password/PasswordFactorySpi.java

```diff
@@ -89,6 +90,20 @@ protected PasswordFactorySpi() {
      */
     protected abstract boolean engineVerify(String algorithm, Password password, final char[] guess) throws InvalidKeyException;
 
+    /**
+     * Perform password verification.
+     *
+     * @param algorithm the password algorithm
+     * @param password the password object
+     * @param guess the guessed password
+     * @param hashCharset the character set used in the password object representation
+     * @return {@code true} if the password matches, {@code false} otherwise
+     * @throws InvalidKeyException if the given password type is not supported
+     */
+    protected boolean engineVerify(String algorithm, Password password, final char[] guess, Charset hashCharset) throws InvalidKeyException {
+        throw new InvalidKeyException();
```

Comment:
This should throw new UnsupportedOperationException() and the following @throws should be added:
@throws UnsupportedOperationException if this method is not supported by the password factory

Reply:
If I change the exception to UnsupportedOperationException, then in the implementation for this method in PasswordFactorySpiImpl, it would have to raise the same exception there. 

Did you mean the method signature should include both InvalidKeyException and UnsupportedOperationException?

Reply:
The body of the method should be changed and an additional @throws should also be added to the Javadoc, e.g.,

```
@throws UnsupportedOperationException if this method is not supported by the password factory implementation
```
The method signature can remain unchanged. Note that `UnsupportedOperationException` is a `RuntimeException`.